Implement an endpoint to update a todo item in the DB by its ID

yamin-faruque-sazim opened the pull request:
**DESCRIPTION OF CHANGES**

- Refactor create-todos-dto to todos.dto, and export UpdateTodosDto from here
- Extract the id from the URL param and body (based on the data shape defined by the UpdateTodosDto) from the request body and send it to the service
- Fetch the todo item based on the id using existing getTodoById endpoint that has validation as well
- Assign any changes if any to the todo item
- Flush

Apologies for the video quality, had to downgrade it as limit is only 10mb

https://github.com/user-attachments/assets/92b89a6c-4459-4038-af65-6beba7beebcb

yamin-faruque-sazim committed:
Improvement(FTDA): Update date validation for createTodos dto

yamin-faruque-sazim committed:
CHORE(FTDA): Refactored create-todos dto to todos dto to include updateTodo dto
FEAT(FTDA): Implement an endpoint to update a todo by its id

ishmam-iqbal-sazim commented on src/todos/todos.service.ts:
```diff
@@ -22,4 +22,11 @@ export class TodosService {
+
+  async updateTodoById(id: string, updateTodo: UpdateTodoDto): Promise<Todo> {
+    const foundTodo = await this.getTodoById(id);
+    this.todoRepository.assign(foundTodo, updateTodo);
+    await this.todoRepository.getEntityManager().flush();
```
In my opinion, this is a case where a repository method (like updateOne) makes sense as the builtin method names are not exactly clear

ishmam-iqbal-sazim commented on src/todos/todos.service.ts:
```diff
@@ -22,4 +22,11 @@ export class TodosService {
     return foundTodo;
   }
+
+  async updateTodoById(id: string, updateTodo: UpdateTodoDto): Promise<Todo> {
```
```suggestion
  async updateOne(id: string, updateTodo: UpdateTodoDto): Promise<Todo> {
```

No need to specify `todo` as this is in todosService.

ishmam-iqbal-sazim commented on src/todos/todos.dto.ts:
```diff
@@ -0,0 +1,59 @@
+  @IsString()
+  description!: string;
+
+  @IsNotEmpty({ message: 'Due date is required.' })
+  @IsDateString({ strict: true })
+  @Matches(/^\d{4}-\d{2}-\d{2}$/, {
+    message: 'Invalid date format. Use YYYY-MM-DD.',
+  })
```
```suggestion
  @IsNotEmpty({ message: 'Due date is required.' })
  @IsDate()
  @Type(() => Date)
```

Can't we do something like this? I am not sure if we need this level of accuracy for dates. Same comment for other instances like this throughout this PR

ishmam-iqbal-sazim reviewed (approved):
Left some comments

yamin-faruque-sazim committed:
CHORE(FTDA): Addressed PR comments:
- update date validation dto
- update service method from updateTodoById to updateOne
- move update logic to todo Repository

yamin-faruque-sazim commented:
Addressed all comments
